src/jit/emitxarch.cpp

```diff
@@ -702,6 +711,22 @@ unsigned emitter::emitOutputRexOrVexPrefixIfNeeded(instruction ins, BYTE* dst, c
         //     VEX.0011RM22 got transformed as VEX.0000RM22
         //
         // Now output VEX prefix leaving the 4-byte opcode
+
+        if ((vexPrefix & 0xFFFF7F80) == 0x00C46100)
```

Comment:
I had initially started doing this by updating AddVexPrefix to support both the 2 and 3-byte encodings and additionally adding a `Supports2ByteVexPrefix` flag to the instruction metadata in `instrsxarch.h`. However, that was leading to a cascade of changes and required knowledge of things like: "is the instruction using any extended registers" to be pre-checked and carried around.

I ended up realizing that, at the point we actually emit the encoding, we have all the relevant information and can just make the decision there, so this ended up being significantly simpler to support than initially expected.

Reply:
That's really great!

Reply:
Shall we add several assertions in this 2-byte VEX encoding section? e.g.,
```cpp
assert(!TakesRexWPrefix(ins));
assert(!IsExtendedReg(reg));
```

Reply:
At this point, we don't have access to the `attr` or the `reg` information, so we can't readily do either of these asserts.

I do think it would be useful to have these types of asserts elsewhere, but I think the only place we logically have all the information is in `emitDispIns`, where we have both the `instrDesc` and the emitted code bytes (but `emitDispIns` needs some of its own cleanup as well). I think this would also be a useful place to have asserts like that `INS_mov_i2xmm` doesn't have an `attr` of `EA_16BYTE` (since one register is implied 16-byte and the attr needs to be used to determine if we are 32-bit or 64-bit instead).
* @CarolEidt, thoughts?

Reply:
In general, I think it's a bad idea to have asserts in `emitDispIns`, at least those that don't directly relate to displaying the instruction. This is because we don't want to have new asserts appearing in a dump or disassembly. Better would be to add an `emitCheckIns` or something, that can always be invoked on the instruction when DEBUG is set.

Reply:
> I think it would be good to handle that in a separate PR

It is okay. I just worried about potential corner case failures, as this is a very fundamental change that impacts all the floating point and SIMD code.